models/badges.js

```diff
@@ -27,6 +27,35 @@ const fetchBadges = async ({
   }
 }
 
+/**
+ * Fetches the data about user badges
+ * @param query { Object }: Filter for badges data
+ * @return {Promise<userBadgeModel|Array>}
+ */
+
+const fetchUserBadges = async (username) => {
+  try {
+    const snapshot = await badgeModel.get()
+    const allBadges = []
+    snapshot.forEach((doc) => {
+      allBadges.push(doc.data())
+    })
+    const userBadges = []
```

Comment:
Do we really need to first insert all data into `allBadges` array and then loop through it and then push it into `userBadges`?

On line no 41, we have the badge data - `doc.data()`
Why not directly check if the userid matches and there itself push it into `userBadges`.

Does this make sense? we can avoid the extra for loop call.

Reply:
Yes it makes sense, I'll make the required changes.